meson.build

```diff
@@ -565,6 +565,13 @@ else
   run_sanitize_unsafe_tests = true
 endif
 
+# take foo.struct and generate foo-struct.c and foo-struct.h files like protobuf_c
+structgenpy = find_program(join_paths(meson.project_source_root(), 'contrib', 'generate-struct.py'))
+structgen = generator(structgenpy,
```

Comment:
for CI purpose, what do you think about having a few bad structs that should fail to process that we test for?

Reply:
Maybe not for CI -- but i certainly found a few converting tons of code ;) The benefit with this generator approach is that any failures to compile are compile **errors**, nor runtime warnings.